common4j/src/main/com/microsoft/identity/common/java/nativeauth/controllers/results/SignUpCommandResult.kt

```diff
@@ -81,60 +98,95 @@ interface SignUpCommandResult {
         SignUpSubmitUserAttributesCommandResult,
-        SignUpSubmitCodeCommandResult
+        SignUpSubmitCodeCommandResult {
+        override fun toUnsanitizedString(): String = "AttributesRequired(correlationId=$correlationId, error=$error, errorDescription=$errorDescription, requiredAttributes=$requiredAttributes)"
+
+        override fun containsPii(): Boolean = true
+
+        override fun toString(): String = "AttributesRequired(correlationId=$correlationId, requiredAttributes=$requiredAttributes)"
+    }
 
     /**
      * The signup operation cannot progress as the provided password is not acceptable
      */
     data class InvalidPassword(
+        override val correlationId: String,
         val error: String,
         val errorDescription: String,
-        override val correlationId: String,
         val subError: String
-    ) : SignUpStartCommandResult, SignUpSubmitPasswordCommandResult
+    ) : SignUpStartCommandResult, SignUpSubmitPasswordCommandResult {
+        override fun toUnsanitizedString(): String = "InvalidPassword(correlationId=$correlationId, error=$error, errorDescription=$errorDescription, subError=$subError)"
+
+        override fun containsPii(): Boolean = true
 
+        override fun toString(): String = "InvalidPassword(correlationId=$correlationId)"
+    }
 
     /**
      * The signup operation cannot progress as the provided out of band code is invalid.
      */
     data class InvalidCode(
+        override val correlationId: String,
         val error: String,
         val errorDescription: String,
-        override val correlationId: String,
         val subError: String
-    ) : SignUpSubmitCodeCommandResult
+    ) : SignUpSubmitCodeCommandResult {
+        override fun toUnsanitizedString(): String = "InvalidCode(correlationId=$correlationId, error=$error, errorDescription=$errorDescription, subError=$subError)"
+
+        override fun containsPii(): Boolean = true
+
+        override fun toString(): String = "InvalidCode(correlationId=$correlationId)"
+    }
 
     /**
      * The signup operation cannot progress as the provided attributes are not valid. Some user
      * attributes can be validated at the server and if the validation fails then this error is returned.
      */
     data class InvalidAttributes(
+        override val correlationId: String,
         val error: String,
         val errorDescription: String,
         val invalidAttributes: List<String>,
-        override val correlationId: String,
-    ) : SignUpStartCommandResult, SignUpSubmitUserAttributesCommandResult
+    ) : SignUpStartCommandResult, SignUpSubmitUserAttributesCommandResult {
+        override fun toUnsanitizedString(): String = "InvalidAttributes(correlationId=$correlationId, error=$error, errorDescription=$errorDescription)"
+
+        override fun containsPii(): Boolean = true
+
+        override fun toString(): String = "InvalidAttributes(correlationId=$correlationId, invalidAttributes=$invalidAttributes)"
```

Comment:
Question: do the invalid attributes returned from the server contain the values in those attributes? If they do, there's potentially sensitive information in there like names, addresses, etc. If it's just the names of the attributes it's probably fine.

Reply:
A few points on this: 
- I believe the server actually doesn't return this response at all. It's a missing feature at the moment.
- Our code assumes that if/when it is returned, the API returns a `List<Map<String, String>>`, of pairs as such: `"name"` - `"<attribute_key>"`. See `internal fun List<Map<String, String>>.toAttributeList(): List<String> `.
- For now, I want to error on the side of caution and consider it PII.
- Which means I need to move it from `toString()` to `toUnsanitisedString()`.